### DISCUSSION TITLE
[논의] BigDecimal와 Money 불변 객체 사용 여부

### DISCUSSION THREAD

**Discussion by june-777** [woowa-techcamp-2024/Team1-3K1K#48] [General]
### 배경

```bash
구매 금액 적립 0.25% 요구사항으로, 금액에 대한 소수점을 고려해야 함
또한 금액의 경우 불변 객체로 설계하여 진행할지 고민이 필요
```

### 세부 주제

```bash
1.
2.
3.
```

**Comment by Dr-KoKo** [comment]
서비스가 고도화되면 결국에는 `Money`와 같은 Embedable이 필요할 것이라고 생각합니다.
단위(원/달러)가 추가될 수도 있고, 소수점도 고려해야할 것 같아요.

**Comment by Hyeon-Uk** [comment]
금액의 경우 Domain객체 내부에서 불변 객체로 존재해야 안전할것 같다고 설득당했지만, `final BigDecimal` 과 같은 타입으로 불변객체로 만들어 버린다면, 도메인 내부에서 수행되는 입/출금, 충전에 대해 내부에서도 값을 변경시킬 방법이 없을것같습니다.

확실히 Money라는 객체로 감싸 Money를 final 로 묶어준 뒤, Money 내부의 메서드를 protected로 접근을 제어해서 안전하게 처리해주는것이 좋을 것 같습니다!
